startup.py

```diff
@@ -139,16 +137,19 @@ def prepare_launch(self, exec_path, args, file_to_open=None):
         for option in options:
             file_dialog.setOption(option)
         # browse!
-        if file_dialog.exec_():
-            selected_file_path = file_dialog.selectedFiles()[0]
-            print("ue_project_dir = {}".format(selected_file_path))
-
-        args += "-projectPath " +  selected_file_path
-        self.logger.debug("Launch environment: %s", pprint.pformat(required_env))
-        self.logger.debug("Launch arguments: %s", args)
-         
-        
-        return LaunchInformation(exec_path, args, required_env)
+        while True and file_dialog.exec_():
```

Comment:
this can be abbreviated to: `while file_dialog.exec_()`

Reply:
If i abbreviate that, then if the user choses the wrong type of file, they'll get an error message but it won't relaunch the file dialog for them again , we want the error message to pop up , they click okay, then file dialog shows up again for them to chose the right type of folder